### DISCUSSION TITLE
[FEATURE REQUEST] Set boot option for VmWare Salt-Cloud

### DISCUSSION THREAD

**Discussion by AntRobinson** [saltstack/salt#67257] [Feature Requests]
I am unable to set the Firmware boot options when creating a VM with Salt-Cloud, with a VM hosted in VMware VSphere. When the VM is created it defaults to BIOS however I would like for it to create the VM with EFI instead.

It would be very useful if there was an option you could add in the .sls cloud profile for "Firmware" (other user my find other boot options to be more applicable so it might be worth looking into that too).

I've tried to use the VMWare extension to get this to work but am yet to get it working. Personally I think it would be better for salt-cloud to handle all of this.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by AntRobinson** [comment]
Did this make it into the Chlorine 3007.0 release?

**Comment by z900collector** [comment]
I'm finding I need this feature as well, is there an implementation and if so, any examples of how to set 'EFI' as a boot pption?
